mono/metadata/marshal-ilgen.c

```diff
@@ -6405,7 +6405,7 @@ emit_native_icall_wrapper_ilgen (MonoMethodBuilder *mb, MonoMethod *method, Mono
 			// }
 			mono_mb_emit_byte (mb, CEE_DUP);
 			int pos = mono_mb_emit_branch (mb, CEE_BRFALSE);
-			mono_mb_emit_ldflda (mb, MONO_HANDLE_PAYLOAD_OFFSET (MonoObject));
+			mono_mb_emit_ldflda (mb, MONO_HANDLE_OFFSET (MonoObject));
```

Comment:
this is really the offset of the handle content in a `MonoRawHandle` - it doesn't have much to do with `MonoObjectHandle`.

Reply:
I was tempted just to say `0` but avoided it.
I guess this was "broken" by typesafe handles in the first place, i.e. it is talking about the wrong level of indirection? Could probably make a struct for that ("Payload" again) and only fix up MONO_HANDLE_RAW. I can try that.

Reply:
Not sure this is worthwhile. Unless maybe we have two of these? Offset of _raw within handle, and offset of another _raw within payload?